docs/root/intro/life_of_a_request.rst

```diff
@@ -0,0 +1,181 @@
+  knowledge of cluster and endpoint health, load balancing and connection pooling exists. The
+  upstream HTTP/2 codec lives here.
+
+The two subsystems are bridged with the HTTP router filter.
+
+We use the labels *ListenerManager* and *ClusterManager* above to refer to the family of modules and
+instance classes that are rooted in these top-level manager classes. In reality, there are numerous
+components that we discuss below that are instantiated prior to and during the course of a request
+by these management systems, for example listeners, filter chains, codecs, connection pools and load
+balancing data structures.
+
+.. note::
+  TODO: add architecture diagram.
+
+Request path
+------------
+
+Overview
+~~~~~~~~
+
+The request path for the initial bytes that constitute the request headers takes the following
+steps:
+
+1. A TCP connection from a downstream is accepted by an Envoy listener associated with a worker
+   thread.
+2. The listener filter chain is instantiated and executed, which can provide SNI and other pre-TLS info.
+3. On network reads, the TLS transport socket transforms the encrypted TCP connection
+   to a plain text network connection.
+4. The network filter chain is instantiated and executed. The most significant filter for HTTP is
+   the HTTP connection manager which sits at the end of the filter chain.
+5. The HTTP/2 codec in HTTP connection manager lifts the plain text TCP connection to a number of
+   independent HTTP streams. Each HTTP stream is a single request.
+6. For each HTTP stream, an HTTP filter chain is instantiated and executed. The most significant
+   filter for HTTP is the router filter which sits at the end of the filter chain. Once
+   *decodeHeaders* processing is completed by the HTTP filter chain, the request headers on the stream
+   are ready to start forwarding to the upstream.
+7. The router filter requests an HTTP connection pool from the cluster manager for the matched
+   cluster.
+8. Cluster specific load balancing is performed to find an endpoint. A new connection to the
+   endpoint is created if the endpoint's connection pool is empty.
+9. The upstream endpoint connection's HTTP/2 codec combines streams from this and other requests into an L4 stream.
+10. The upstream endpoint connection's TLS transport socket encrypts these bytes and writes them to
+    a TCP socket for the upstream connection.
+
+.. note::
+  TODO: high-level request path diagam
+
+Following the request headers, the path setup above processes reads and write events from
+the downstream and upstream endpoints on the TCP connection's worker thread. The request
+will go through a lifecycle which includes both request and response header processing,
+body streaming and trailer processing. Following the completion of a request, post-request
+processing will update stats and write to the access log.
+
+The above description is terse, we elaborate on each of these request processing steps below.
```

Comment:
Maybe call that out at the top, so folks know it's an overview and there's detailed explanation below?